File: GXpert/GXpert.Web/Modules/Analytics/ExamAttempt/ExamAttemptForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Analytics.Forms;

[FormScript("Analytics.ExamAttempt")]
[BasedOnRow(typeof(ExamAttemptRow), CheckNames = true)]
public class ExamAttemptForm
{
    [HalfWidth]
    public int ExamId { get; set; }
    [HalfWidth]
    public int StudentId { get; set; }
    [HalfWidth]
    public int TeacherId { get; set; }
    [HalfWidth]
    public short EStatus { get; set; }
    [HalfWidth]
    public int TimeSpent { get; set; }
    [HalfWidth]
    public int PlayListId { get; set; }
    [HalfWidth]
    public string StudentAnswerSheetUpload { get; set; }
    [HalfWidth]
    public string TeacherCheckedPaperUpload { get; set; }
    [HalfWidth]
    public int ActivationId { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Analytics/ExamAttemptForm.ts
import { LookupEditor, EnumEditor, IntegerEditor, StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { EExamAttemptStatus } from "../Web/Enums.EExamAttemptStatus";

export interface ExamAttemptForm {
    ExamId: LookupEditor;
    StudentId: LookupEditor;
    TeacherId: LookupEditor;
    EStatus: EnumEditor;
    TimeSpent: IntegerEditor;
    PlayListId: LookupEditor;
    StudentAnswerSheetUpload: StringEditor;
    TeacherCheckedPaperUpload: StringEditor;
    ActivationId: LookupEditor;
}

export class ExamAttemptForm extends PrefixedContext {
    static readonly formKey = 'Analytics.ExamAttempt';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ExamAttemptForm.init)  {
            ExamAttemptForm.init = true;

            var w0 = LookupEditor;
            var w1 = EnumEditor;
            var w2 = IntegerEditor;
            var w3 = StringEditor;

            initFormType(ExamAttemptForm, [
                'ExamId', w0,
                'StudentId', w0,
                'TeacherId', w0,
                'EStatus', w1,
                'TimeSpent', w2,
                'PlayListId', w0,
                'StudentAnswerSheetUpload', w3,
                'TeacherCheckedPaperUpload', w3,
                'ActivationId', w0
            ]);
        }
    }
}

[EExamAttemptStatus]; // referenced types